### PR TITLE
mrc-2670 add warnings to module states

### PR DESCRIPTION
**Pull request by hillalex**
## Description

* Re-generated types using this branch of hintr: https://github.com/mrc-ide/hintr/pull/304
 Note hintr master won't actually be returning these warning properties, but as the compiled js doesn't care about undefined properties and we're not yet surfacing them in the UI this won't cause any errors
* Added `warnings` as a property on `modelOptions`, `modelRun` and `modelCalibrate` module states, and populated these when present in response objects - see https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-2664 for the complete list of such objects. ~~Note that although 3 endpoints return warnings, there are only 2 mutations that need updating, because the same mutation is invoked by 2 different actions.~~

I think this is all that's required as far as getting warnings into the app state goes.

## Type of version change
Patch

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

File: src/app/static/src/app/store/modelRun/mutations.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ export const mutations: MutationTree<ModelRunState> = {
 
     [ModelRunMutation.RunResultFetched](state: ModelRunState, action: PayloadWithType<ModelResultResponse>) {
         state.result = action.payload;
+        state.warnings = action.payload.warnings
```
Comment (hillalex) [inline src/app/static/src/app/store/modelRun/mutations.ts:44]: Note this mutation is invoked twice - [once during fitting](https://github.com/mrc-ide/hint/blob/master/src/app/static/src/app/store/modelRun/actions.ts#L48) and [once during calibration](https://github.com/mrc-ide/hint/blob/master/src/app/static/src/app/store/modelCalibrate/actions.ts#L74)

File: src/app/static/src/app/generated.d.ts [hunk 2]
```diff
@@ -400,6 +528,10 @@ export interface ModelCalibrateRequest {
 export type ModelCancelResponse = null;
 export interface ModelOptionsValidate {
   valid: true;
+  warnings: {
+    text: string;
+    locations: ("model_options" | "model_fit" | "model_calibrate" | "review_output" | "download_results")[];
+  }[];
```
Comment (Nicolas-Dolan) [inline src/app/static/src/app/generated.d.ts:534]: I'm learning new ts here. I guess this signifies an array of locations, containing one or more of the above string values?

File: src/app/static/src/app/store/modelOutput/modelOutput.ts [hunk 3]
```diff
@@ -66,7 +66,8 @@ const outputPlotFilters = (rootState: RootState) => {
 
 export const initialModelOutputState = (): ModelOutputState => {
     return {
-        selectedTab: ""
+        selectedTab: "",
+        warnings: []
```
[the remark this answers is not in the document]
Comment (hillalex) [reply src/app/static/src/app/store/modelOutput/modelOutput.ts:70]: You're right, this is a superfluous

**Review by mwoodbri — APPROVED** [review]
Looks good - just one superfluous change that would ideally be removed before merge.

File: src/app/static/src/app/store/modelOutput/modelOutput.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import {Module} from "vuex";
-import {RootState} from "../../root";
+import {RootState, WarningsState} from "../../root";
```
[the remark this answers is not in the document]
Comment (hillalex) [reply src/app/static/src/app/store/modelOutput/modelOutput.ts]: https://github.com/mrc-ide/hint/pull/568/commits/398e3e8ba7b7a2e00df7655386a1e652ae44c2b5

**Review by carbowizzy — APPROVED** [review]

**Pull request merged**